adafruit_register/spi_bit.py

```diff
@@ -0,0 +1,83 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2016 Scott Shawcroft for Adafruit Industries
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+# pylint: disable=too-few-public-methods
+"""
+`adafruit_register.spi_bit`
+====================================================
+
+Single bit registers
+
+* Author(s): Scott Shawcroft
+* Adaptation by Max Holliday
+"""
+
+
+class RWBit:
+    """
+    Single bit register that is readable and writeable.
+
+    Values are `bool`
+
+    :param int register_address: The register address to read the bit from
+    :param type bit: The bit index within the byte at ``register_address``
+    :param int register_width: The number of bytes in the register. Defaults to 1.
+    :param bool lsb_first: Is the first byte we read from spi the LSB? Defaults to true
+
+    """
+    def __init__(self, register_address, bit, register_width=1, lsb_first=True):
```

Comment:
@siddacious Could you elaborate and give an example? I'm not sure of the context here.

Reply:
@siddacious Can you please follow up on this PR? Thanks!

Reply:
@kattni I'm going to finally get back to this and address @tannewt's request for more info!

Reply:
@tannewt to elaborate:
SPI uses  a bit within the first byte sent to designate if the following bytes will be read or written by the host. As far as I know, it's up to the manufacturer to decide if a high level designates a read or a write. Here is the relevant section of the datasheet for the bosch IMU mentioned by Dr. M:
![image](https://user-images.githubusercontent.com/1183985/94847167-12037300-03d7-11eb-8e43-c5c89cbff0ea.png)

I'm suggesting the polarity of this should be able to be set ~~in the constructor of `busio.SPI`~~, with a default to `rw_high_write=True` as it is the most common. 
edit: see below

Additionally, SPI devices often use a bit in the address byte to tell the sensor to automatically increment the register address for successive reads, which allows you to read contiguous registers without specifying the address of the next register each time. Getting this right is particularly important because if it isn't set correctly, you can still get data back that seems valid but that will just be repeats of the data from the register with the specified address.

Here is an example from the bosch imu datasheet that implies that such a bit isn't needed:
![image](https://user-images.githubusercontent.com/1183985/94848074-504d6200-03d8-11eb-8b62-372c2e822126.png)

And one from the LIS3DH that explains that it is:
![image](https://user-images.githubusercontent.com/1183985/94848171-75da6b80-03d8-11eb-8de8-7c610654a26c.png)

The proper setting of these bits is necessary for the proper functioning of the sensor library, and since the setting will always be the same for a given sensor, ideally it would be able to be set once. I was originally suggesting that this should be set in the SPI constructor, however since an SPI bus can be shared, it has the potential to cause issues if some of the sensors on the bus have different settings. 

That said, the clock polarity and phase can be different between sensors and it is set in the `busio.SPI` object. It's not clear to me at this point where the ideal place to set this would be. Perhaps something like a `Register` factory object that is then used to create the individual `RWBits`, etc.

I've had this ball in my court for a while now and I'd like to see SPI support in `Register`, so any input that will help us make a choice and move forward would be appreciated.


PS: here's a graphic that explains the SPI clock polarity and phase settings for those who want more info:

from: https://en.wikipedia.org/wiki/Serial_Peripheral_Interface#Clock_polarity_and_phase
![image](https://user-images.githubusercontent.com/1183985/94849514-62c89b00-03da-11eb-91f0-a097410b0f58.png)

Reply:
@lavericklavericklaverick ☝️

Reply:
@siddacious I think the simplest thing is to have it as part of the register field's constructor. Then it can change the top bit or two as needed. In the longer term it'd be interesting to unify the bus API so we can have a device with registers accessible over i2c or spi.

Reply:
fair enough

Reply:
@tannewt to keep this moving, should I look into adding the RW/multi-byte bit behavior?

Reply:
@siddacious If I remember right, you'll want two kwargs. One which is the bit position to indicate read vs write and then one bit position to indicate no address increment.